### PR TITLE
feat: hover effect in vt mode

### PR DESCRIPTION
**Pull request by cbt1**
Enables hover effect in vt mode.

As no "rows" exist in vt mode. Hover effect is applied using a state shared by all cells and mouse events on cell level that update the state. My main concern with that was performance but it seems good enough.

### CODE REVIEW

File: src/table/components/virtualized-table/utils/__tests__/get-cell-style.test.ts [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+        background:
+          'repeating-linear-gradient(-45deg, rgba(200,200,200,0.08), rgba(200,200,200,0.08) 2px, rgba(200,200,200,0.3) 2.5px, rgba(200,200,200,0.08) 3px, rgba(200,200,200,0.08) 5px), bodyBackground',
+        borderColor: 'bodyBorderColor',
+        hoverBackgroundColor: 'hoverBackgroundColor',
+        hoverFontColor: 'hoverFontColor',
+      });
+    });
+  });
+
+  describe('user is NOT hovering row', () => {
```
Comment (haxxmaxx) [inline src/table/components/virtualized-table/utils/__tests__/get-cell-style.test.ts:81]: I would say it is a bit over-kill to test all cases for both hovering off and not hovering, but it is fine

File: src/table/components/virtualized-table/TableContainer.tsx [hunk 2]
```diff
@@ -22,13 +22,12 @@ const TableContainer = (props: TableContainerProps) => {
...
-  const bodyStyle = useMemo(
+  const bodyStyle = useMemo<BodyStyle>(
     () => ({
       ...getBodyCellStyle(layout, theme),
-      backgroundColor: theme.background.color, // Append both background and backgroundColor to avoid conflicting prop error when selecting styling is applied
-      background: theme.background.color,
+      background: theme.background.color ?? 'transparent',
     }),
-    [layout, theme]
+    [layout, theme.name()] // eslint-disable-line react-hooks/exhaustive-deps
```
[the remark this answers is not in the document]
Comment (cbt1) [reply src/table/components/virtualized-table/TableContainer.tsx:30]: Yeah we do

File: src/table/components/virtualized-table/utils/__tests__/get-cell-style.test.ts (hunk 1 above)
Comment (cbt1) [reply src/table/components/virtualized-table/utils/__tests__/get-cell-style.test.ts:81]: I would agree. I only had 1 test initial to ensure branch coverage, but ended up thinking that I might as well add for "all" scenarios

File: src/table/components/virtualized-table/utils/get-cell-style.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import { BodyStyle } from '../types';
+
+const getCellStyle = (
+  showHoverEffect: boolean,
+  hoverIndex: number,
+  rowIndex: number,
+  cellSelectionState: SelectionStates,
+  bodyStyle: BodyStyle
+): BodyStyle => {
+  const isHoveringOnRow = showHoverEffect && hoverIndex === rowIndex;
```
[the remark this answers is not in the document]
Comment (cbt1) [reply src/table/components/virtualized-table/utils/get-cell-style.ts:11]: I think I can refactor, that would also reduce the number of scenarios in `getCellStyle` tests

[the remark this answers is not in the document]
Comment (cbt1) [reply src/table/components/virtualized-table/utils/get-cell-style.ts:11]: Refactored

**Review by a-m-dev — APPROVED** [review]
LGTM, 
Regarding performance aspect, My suggestions would be to limit the resources and then check it with a (semi-)heavy data set. If you see some bottlenecks => maybe it worth to consider ref trick (ref as state holder) in order to keep track of changed index which would cause no rerender of table tree.

**Pull request merged**